java/src/main/c++/vw_learner_VWFloatArrayLearner.cc

```diff
@@ -3,9 +3,9 @@
 #include "jni_base_learner.h"
 #include "vw_learner_VWFloatArrayLearner.h"
 
-jfloatArray floatArrayPredictor(example* vec, JNIEnv *env) {
+jfloatArray floatArrayPredictor(example* vec, JNIEnv *env, jlong vwPtr) {
     v_array<float> predictions = vec->topic_predictions;
-    size_t num_predictions = predictions.size();
+    size_t num_predictions = ((vw*)vwPtr)->lda;
```

Comment:
Since we already have done this cast in jni_base_learner.h it seems silly to do it here too.  Let's have predictor accept the vwInstance directly instead of the pointer.  That would require minor changes to all the files, but it should be better cause you don't do this cast twice.